homeWork4/lesson1/lesson1/JsonDownloader.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+namespace lesson1;
+
+public class JsonDownloader
+{
+	public JsonDownloader()
+    {
+		_myClient = new HttpClient();
+	}
+
+	private HttpClient _myClient;
+
+	public HttpClient MyClient
+    {
+		get => _myClient;
+		set => _myClient = value;
+    }
+
+	public async Task<T> Start<T>(string link, int id)
+	{		
+		CancellationTokenSource tokenSource = new CancellationTokenSource();

```

Comment:
```suggestion
		using CancellationTokenSource tokenSource = new CancellationTokenSource();

```

тогда ниже не надо делать `finally`